server/routes/updateGoal/updateGoalController.ts

```diff
@@ -41,10 +41,11 @@ export default class UpdateGoalController {
   submitUpdateGoalForm: RequestHandler = async (req, res, next): Promise<void> => {
     const { prisonNumber, goalReference } = req.params
     const updateGoalForm: UpdateGoalForm = { ...req.body }
+    const updateStepForm: UpdateStepForm = { ...req.body }
     req.session.updateGoalForm = updateGoalForm
 
     const errors = validateUpdateGoalForm(updateGoalForm)
-    if (errors.length > 0) {
+    if (errors.length > 0 && !updateStepForm.action.includes('delete-step-')) {
```

Comment:
This seems to be breaking the integration tests for submitting the page with validation issues, however the Cypress tests are passing fine for submitting the page with validation issues.

Reply:
This is to stop the validation of empty form fields when the Remove step button is clicked. A user could click "Add another step" but then decide they don't need that step anymore, but because the Step data is blank, they wouldn't be able to remove it.